etc/telegraf/telegraf.toml.j2

```diff
@@ -238,7 +238,7 @@
 {% endif %}
 {% endif %}
 
-{% if datadog_api_key and db_config and database_diskstorage_metric_enabled %}
+{% if (datadog_api_key or (appdynamics_enabled and machine_agent_enabled)) and db_config and database_diskstorage_metric_enabled %}
```

Comment:
Isn't the `db_config` check enough and we get the database metrics when `db_config` is enabled ?

Reply:
Some additional checks for appdynamics added in to "db_config" check.